### DISCUSSION TITLE
Signing request confusion

### DISCUSSION THREAD

**Discussion by nemishzalavadiya-pirimid** [ONDC-Official/developer-docs#32] [Q&A] [answered]
here [Signing Beckn APIs in HTTP
](https://docs.google.com/document/d/1Iw_x-6mtfoMh0KJwL4sqQYM0kD17MLxiMCUOZDBerBo/edit)

At 4th step, please confirm

> Concatenate the three values, i.e the `created`, `expires` and `digest` in the format as shown below. The below string is the signing string which the BAP is going to use to sign the request

Request data format is this,

(created): 1641287875
(expires): 1641291475
digest: BLAKE-512=b6lf6lRgOweajukcvcLsagQ2T60+85kRh/Rd2bdS+TG/5ALebOEgDJfyCrre/1+BMu5nA94o4DT3pTFXuUg7sw==

or this,

(created): 1641287875
(expires): 1641291475
digest: 
BLAKE-512=b6lf6lRgOweajukcvcLsagQ2T60+85kRh/Rd2bdS+TG/5ALebOEgDJfyCrre/1+BMu5nA94o4DT3pTFXuUg7sw==
?

Also Please confirm `Concatenate the three values` is default concat with newline character \n?

**Comment by 92shreyansh** [comment, marked as answer]
signString = "(created): " + currentTime + "\n(expires): " + (currentTime + VALIDITY)+ "\ndigest: BLAKE-512=" + blakeHash + "";
